<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>consensus. bio3d 2.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content=" Barry Grant ">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">

<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="http://thegrantlab.org/bio3d/index.html">bio3d 2.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Index</a></li>
          <li><a href="http://thegrantlab.org/bio3d/download/download.html">Download Bio3D</a></li>
          <li><a href="http://thegrantlab.org">The Grant Lab</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h1> Sequence Consensus for an Alignment </h1>

<div class="row">
  <div class="span8">
    <h2>Usage</h2>
    <pre>consensus(alignment, cutoff = 0.6)</pre>
    
    <h2>Arguments</h2>
    <dl>
      <dt>alignment</dt>
      <dd> an <code>alignment</code> object created by the
    <code><a href='read.fasta.html'>read.fasta</a></code> function or an alignment character matrix. </dd>
      <dt>cutoff</dt>
      <dd> a numeric value beteen 0 and 1, indicating the minimum
    sequence identity threshold for determining a consensus amino acid.
    Default is 0.6, or 60 percent residue identity. </dd>
    </dl>
    
    <div class="Description">
      <h2>Description</h2>
      
      <p>Determines the consensus sequence for a given alignment at a given
  identity cutoff value.</p>
  
    </div>

    <div class="Value">
      <h2>Value</h2>
      
      <p>A vector containing the consensus sequence, where &#145;-&#146; represents
  positions with no consensus (i.e. under the <code>cutoff</code>)</p>
  
    </div>

    <div class="References">
      <h2>References</h2>
      
      <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
  
    </div>
    
    <h2 id="examples">Examples</h2>
    <pre class="examples"><div class='input'>
#-- Read HIV protease alignment
aln <- read.fasta(system.file("examples/hivp_xray.fa",package="bio3d"))

# Generate consensus
con <- consensus(aln)
print(con$seq)
</div>
<div class='output'>  [1] "P" "Q" "I" "T" "L" "W" "Q" "R" "P" "L" "V" "T" "I" "K" "I" "G" "G" "Q"
 [19] "L" "K" "E" "A" "L" "L" "D" "T" "G" "A" "D" "D" "T" "V" "L" "E" "E" "M"
 [37] "S" "L" "P" "G" "R" "W" "K" "P" "K" "M" "I" "G" "G" "I" "-" "-" "G" "G"
 [55] "F" "I" "K" "V" "R" "Q" "Y" "D" "Q" "I" "-" "I" "E" "I" "C" "G" "H" "K"
 [73] "A" "I" "G" "T" "V" "L" "V" "G" "P" "T" "P" "V" "N" "I" "I" "G" "R" "N"
 [91] "L" "L" "T" "Q" "I" "G" "C" "T" "L" "N" "F"
</div>
<div class='input'>
# Plot residue frequency matrix
##png(filename = "freq.png", width = 1500, height = 780)
col <- mono.colors(32)
aa  <- rev(rownames(con$freq))

image(x=1:ncol(con$freq),
      y=1:nrow(con$freq),
      z=as.matrix(rev(as.data.frame(t(con$freq)))),
      col=col, yaxt="n", xaxt="n",
      xlab="Alignment Position", ylab="Residue Type")
</div>
<p><img src='consensus-4.png' alt='' width='400' height='400' /></p>
<div class='input'>
# Add consensus along the axis
axis(side=1, at=seq(0,length(con$seq),by=5))
</div>
<p><img src='consensus-6.png' alt='' width='400' height='400' /></p>
<div class='input'>axis(side=2, at=c(1:22), labels=aa)
</div>
<p><img src='consensus-8.png' alt='' width='400' height='400' /></p>
<div class='input'>axis(side=3, at=c(1:length(con$seq)), labels =con$seq)
</div>
<p><img src='consensus-10.png' alt='' width='400' height='400' /></p>
<div class='input'>axis(side=4, at=c(1:22), labels=aa)
</div>
<p><img src='consensus-12.png' alt='' width='400' height='400' /></p>
<div class='input'>grid(length(con$seq), length(aa))
</div>
<p><img src='consensus-14.png' alt='' width='400' height='400' /></p>
<div class='input'>box()
</div>
<p><img src='consensus-16.png' alt='' width='400' height='400' /></p>
<div class='input'>
# Add consensus sequence
for(i in 1:length(con$seq)) {
  text(i, which(aa==con$seq[i]),con$seq[i],col="white")
}
</div>
<p><img src='consensus-18.png' alt='' width='400' height='400' /></p>
<div class='input'>
# Add lines for residue type separation
abline(h=c(2.5,3.5, 4.5, 5.5, 3.5, 7.5, 9.5,
         12.5, 14.5, 16.5, 19.5), col="gray")
</div>
<p><img src='consensus-20.png' alt='' width='400' height='400' /></p>
<div class='input'></div></pre>
  </div>
  <div class="span4">
    <!-- <ul>
      <li>consensus</li>
    </ul>
    <ul>
      <li> utilities </li>
    </ul> -->
      
    <h2>See also</h2>
     <code><a href='read.fasta.html'>read.fasta</a></code> 
    
    <h2>Author</h2>
     Barry Grant 
    
  </div>
</div>
      
      <footer>
      <p class='feedback'>What do you think of the documentation?  <a href='http://xinqyao.wufoo.com/forms/documentation-feedback-for-bio3d/def/field1=consensus'>Please let me know by filling out this short online survey</a>.</p>
<p class="pull-right"><a href="#">Back to top</a></p>
<p>Built by <a href="https://github.com/hadley/staticdocs">staticdocs</a>. Styled with <a href="http://twitter.github.com/bootstrap">bootstrap</a>.</p>
      </footer>
    </div>
  </body>
</html>
