<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Index. bio3d 2.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Barry Grant, Xin-Qiu Yao, Lars Skjaerven">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">

<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="http://thegrantlab.org/bio3d/index.html">bio3d 2.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Index</a></li>
          <li><a href="http://thegrantlab.org/bio3d/download/download.html">Download Bio3D</a></li>
          <li><a href="http://thegrantlab.org">The Grant Lab</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <div class="row">
  <div class="span8">
    <div class="hero-unit">
<h1>Bio3D in R</h1>

<p>Utilities to analyze, process, organize and explore protein structure, sequence and dynamics data.</p>
<p><a href="http://thegrantlab.org/bio3d/download/download.html" class="btn btn-primary btn-large">Download now &raquo;</a></p>  
</div>

<h1>Features</h1>

<p>Features include the ability to read and write structure, sequence and dynamic trajectory data, perform database searches, atom summaries, atom selection, re-orientation, superposition, rigid core identification, clustering, torsion analysis, distance matrix analysis, structure and sequence conservation analysis, and principal component analysis (PCA).  </p>

<p>In addition, various utility functions are provided to enable the statistical and graphical power of the R environment to work with biological sequence and structural data.  Please refer to the main <code>Bio3D website</code> at <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a> for more background information.</p>

<p>The Bio3D package consists of input/output functions, conversion and manipulation functions, analysis functions, and graphics functions which are summarized below with links to further documentation including example results. Remember that you can also get help on any particular function by using the command <code>?function</code> or <code>help(function)</code> from within R itself.</p>


    <h2>Help topics</h2>

    <h3>Input/Output:</h3>
    <p>Read and Write Common Data Types</p>

    
    <ul class="index">
        
      <li>
        <code><a href="aln2html.html">aln2html</a></code>: Create a HTML Page For a Given Alignment </li>
            
      <li>
        <code><a href="get.pdb.html">get.pdb</a></code>: Download PDB Coordinate Files </li>
            
      <li>
        <code><a href="get.seq.html">get.seq</a></code>: Download FASTA Sequence Files </li>
            
      <li>
        <code><a href="mktrj.pca.html">mktrj.pca</a></code>: PCA Atomic Displacement Trajectory </li>
            
      <li>
        <code><a href="read.all.html">read.all</a></code>: Read Aligned Structure Data</li>
            
      <li>
        <code><a href="read.crd.html">read.crd</a></code>: Read CRD File </li>
            
      <li>
        <code><a href="read.dcd.html">read.dcd</a></code>: Read CHARMM/X-PLOR/NAMD Binary DCD files </li>
            
      <li>
        <code><a href="read.fasta.html">read.fasta</a></code>: Read FASTA formated Sequences </li>
            
      <li>
        <code><a href="read.fasta.pdb.html">read.fasta.pdb</a></code>: Read Aligned Structure Data </li>
            
      <li>
        <code><a href="read.ncdf.html">read.ncdf</a></code>: Read AMBER Binary netCDF files </li>
            
      <li>
        <code><a href="read.pdb.html">read.pdb</a></code>(print.pdb, summary.pdb): Read PDB File </li>
            
      <li>
        <code><a href="read.pdcBD.html">read.pdcBD</a></code>: Read PQR output from pdcBD File </li>
            
      <li>
        <code><a href="read.pqr.html">read.pqr</a></code>: Read PQR File </li>
            
      <li>
        <code><a href="write.crd.html">write.crd</a></code>: Write CRD File </li>
            
      <li>
        <code><a href="write.fasta.html">write.fasta</a></code>: Write FASTA Formated Sequences </li>
            
      <li>
        <code><a href="write.ncdf.html">write.ncdf</a></code>: Write AMBER Binary netCDF files </li>
            
      <li>
        <code><a href="write.pdb.html">write.pdb</a></code>: Write PDB Format Coordinate File </li>
            
      <li>
        <code><a href="write.pqr.html">write.pqr</a></code>: Write PQR Format Coordinate File </li>
    
    </ul>
    <h3>Analysis:</h3>
    <p>Do Interesting Things</p>

    
    <ul class="index">
        
      <li>
        <code><a href="angle.xyz.html">angle.xyz</a></code>: Calculate the Angle Between Three Atoms </li>
            
      <li>
        <code><a href="blast.pdb.html">blast.pdb</a></code>: NCBI BLAST Sequence Search </li>
            
      <li>
        <code><a href="cmap.html">cmap</a></code>: Contact Map </li>
            
      <li>
        <code><a href="consensus.html">consensus</a></code>: Sequence Consensus for an Alignment </li>
            
      <li>
        <code><a href="conserv.html">conserv</a></code>: Score Residue Conservation At Each Position in an Alignment </li>
            
      <li>
        <code><a href="core.find.html">core.find</a></code>: Identification of Invariant Core Positions </li>
            
      <li>
        <code><a href="dccm.html">dccm</a></code>: DCCM: Dynamical Cross-Correlation Matrix </li>
            
      <li>
        <code><a href="dist.xyz.html">dist.xyz</a></code>:  Calculate the Distances Between the Rows of Two Matrices </li>
            
      <li>
        <code><a href="dm.html">dm</a></code>(dm.xyz): Distance Matrix Analysis </li>
            
      <li>
        <code><a href="dssp.html">dssp</a></code>(stride, print.sse): Secondary Structure Analysis with DSSP or STRIDE </li>
            
      <li>
        <code><a href="entropy.html">entropy</a></code>: Shannon Entropy Score </li>
            
      <li>
        <code><a href="fit.xyz.html">fit.xyz</a></code>(rot.lsq): Coordinate Superposition </li>
            
      <li>
        <code><a href="ide.filter.html">ide.filter</a></code>: Percent Identity Filter </li>
            
      <li>
        <code><a href="seqidentity.html">seqidentity</a></code>: Percent Identity</li>
            
      <li>
        <code><a href="mktrj.pca.html">mktrj.pca</a></code>: PCA Atomic Displacement Trajectory </li>
            
      <li>
        <code><a href="motif.find.html">motif.find</a></code>: Find Sequence Motifs. </li>
            
      <li>
        <code><a href="overlap.html">overlap</a></code>: Overlap analysis </li>
            
      <li>
        <code><a href="pca.project.html">pca.project</a></code>(pca.z2xyz, pca.xyz2z): Project Data onto Principal Components </li>
            
      <li>
        <code><a href="pca.tor.html">pca.tor</a></code>: Principal Component Analysis </li>
            
      <li>
        <code><a href="pca.xyz.html">pca.xyz</a></code>: Principal Component Analysis </li>
            
      <li>
        <code><a href="pdbaln.html">pdbaln</a></code>: Sequence Alignment of PDB Files </li>
            
      <li>
        <code><a href="rgyr.html">rgyr</a></code>: Radius of Gyration</li>
            
      <li>
        <code><a href="rmsd.html">rmsd</a></code>: Root Mean Square Deviation </li>
            
      <li>
        <code><a href="rmsd.filter.html">rmsd.filter</a></code>: RMSD Filter </li>
            
      <li>
        <code><a href="rmsf.html">rmsf</a></code>: Atomic RMS Fluctuations </li>
            
      <li>
        <code><a href="rmsip.html">rmsip</a></code>: Root Mean Square Inner Product </li>
            
      <li>
        <code><a href="seq2aln.html">seq2aln</a></code>: Add a Sequence to an Existing Alignmnet </li>
            
      <li>
        <code><a href="seqaln.html">seqaln</a></code>: Sequence Alignment with MUSCLE</li>
            
      <li>
        <code><a href="seqaln.pair.html">seqaln.pair</a></code>: Sequence Alignment of Identical Protein Sequences </li>
            
      <li>
        <code><a href="seqbind.html">seqbind</a></code>: Combine Sequences by Rows Without Recycling </li>
            
      <li>
        <code><a href="torsion.pdb.html">torsion.pdb</a></code>: Calculate Mainchain and Sidechain Torsion/Dihedral Angles </li>
            
      <li>
        <code><a href="torsion.xyz.html">torsion.xyz</a></code>: Calculate Torsion/Dihedral Angles </li>
            
      <li>
        <code><a href="wrap.tor.html">wrap.tor</a></code>: Wrap Torsion Angle Data </li>
    
    </ul>
    <h3>Graphics:</h3>
    <p>Plotting and Graphic Display</p>

    
    <ul class="index">
        
      <li>
        <code><a href="bwr.colors.html">bwr.colors</a></code>(mono.colors): Color Palettes </li>
            
      <li>
        <code><a href="plot.bio3d.html">plot.bio3d</a></code>: Plots with marginal SSE annotation </li>
            
      <li>
        <code><a href="plot.blast.html">plot.blast</a></code>: Plot a Summary of BLAST Hit Statistics. </li>
            
      <li>
        <code><a href="plot.core.html">plot.core</a></code>: Plot Core Fitting Progress </li>
            
      <li>
        <code><a href="plot.dccm.html">plot.dccm</a></code>(plot.dccm2): DCCM Plot </li>
            
      <li>
        <code><a href="plot.dmat.html">plot.dmat</a></code>: Plot Distance Matrix </li>
            
      <li>
        <code><a href="plot.pca.html">plot.pca</a></code>(plot.pca.score, plot.pca.scree): Plot PCA Results </li>
            
      <li>
        <code><a href="plot.pca.loadings.html">plot.pca.loadings</a></code>:  Plot Residue Loadings along PC1 to PC3 </li>
    
    </ul>
    <h3>Utilities:</h3>
    <p>Convert and Manipulate Data</p>

    
    <ul class="index">
        
      <li>
        <code><a href="aa.index.html">aa.index</a></code>: AAindex: Amino Acid Index Database </li>
            
      <li>
        <code><a href="aa123.html">aa123</a></code>(aa321): Convert Between 1-letter and 3-letter Aminoacid Codes </li>
            
      <li>
        <code><a href="aa2index.html">aa2index</a></code>: Convert an Aminoacid Sequence to AAIndex Values </li>
            
      <li>
        <code><a href="aln2html.html">aln2html</a></code>: Create a HTML Page For a Given Alignment </li>
            
      <li>
        <code><a href="atom.select.html">atom.select</a></code>: Atom Selection From PDB Structure </li>
            
      <li>
        <code><a href="atom2xyz.html">atom2xyz</a></code>(xyz2atom): Convert Between Atom and xyz Indices </li>
            
      <li>
        <code><a href="bio3d.package.html">bio3d-package</a></code>(bio3d):
</li>
            
      <li>
        <code><a href="bounds.html">bounds</a></code>: Bounds of a Numeric Vector </li>
            
      <li>
        <code><a href="chain.pdb.html">chain.pdb</a></code>: Find Possible PDB Chain Breaks </li>
            
      <li>
        <code><a href="convert.pdb.html">convert.pdb</a></code>:Convert Between Various PDB formats</li>
            
      <li>
        <code><a href="diag.ind.html">diag.ind</a></code>: Diagonal Indices of a Matrix </li>
            
      <li>
        <code><a href="difference.vector.html">difference.vector</a></code>: Difference Vector </li>
            
      <li>
        <code><a href="gap.inspect.html">gap.inspect</a></code>: Alignment Gap Summary </li>
            
      <li>
        <code><a href="ide.filter.html">ide.filter</a></code>: Percent Identity Filter </li>
            
      <li>
        <code><a href="is.gap.html">is.gap</a></code>: Gap Characters </li>
            
      <li>
        <code><a href="lbio3d.html">lbio3d</a></code>: List all Functions in the bio3d Package </li>
            
      <li>
        <code><a href="orient.pdb.html">orient.pdb</a></code>: Orient a PDB Structure </li>
            
      <li>
        <code><a href="pairwise.html">pairwise</a></code>: Pair Indices </li>
            
      <li>
        <code><a href="plot.bio3d.html">plot.bio3d</a></code>: Plots with marginal SSE annotation </li>
            
      <li>
        <code><a href="print.core.html">print.core</a></code>: Printing Core Positions and Returning Indices </li>
            
      <li>
        <code><a href="rle2.html">rle2</a></code>(print.rle2):Run Length Encoding with Indices</li>
            
      <li>
        <code><a href="rmsd.filter.html">rmsd.filter</a></code>: RMSD Filter </li>
            
      <li>
        <code><a href="pdbseq.html">pdbseq</a></code>: Extract The Aminoacid Sequence From A PDB Object </li>
            
      <li>
        <code><a href="seqbind.html">seqbind</a></code>: Combine Sequences by Rows Without Recycling </li>
            
      <li>
        <code><a href="pdbsplit.html">pdbsplit</a></code>: Split a PDB File Into Separate Files, One For Each Chain. </li>
            
      <li>
        <code><a href="store.atom.html">store.atom</a></code>: Store all-atom data from a PDB object </li>
            
      <li>
        <code><a href="trim.pdb.html">trim.pdb</a></code>: Trim a PDB Object To A Subset of Atoms. </li>
            
      <li>
        <code><a href="unbound.html">unbound</a></code>: Sequence Generation from a Bounds Vector </li>
            
      <li>
        <code><a href="vec2resno.html">vec2resno</a></code>: Replicate Per-residue Vector Values </li>
    
    </ul>
    <h3>Example Data:</h3>
    <p>Bio3d Example Data</p>

    
    <ul class="index">
        
      <li>
        <code><a href="example.data.html">example.data</a></code>(kinesin, transducin, pdbs, core, annotation):Bio3d Example Data</li>
    
    </ul>
    <h3>Other</h3>
    
    
    <ul class="index">
        
      <li>
        <code><a href="aa2mass.html">aa2mass</a></code>: Amino Acid Residues to Mass Converter </li>
            
      <li>
        <code><a href="atom.index.html">atom.index</a></code>: Index of Atomic Masses </li>
            
      <li>
        <code><a href="atom2mass.html">atom2mass</a></code>(atom2ele, formula2mass): Atom Names to Mass Converter </li>
            
      <li>
        <code><a href="binding.site.html">binding.site</a></code>: Binding Site Residues </li>
            
      <li>
        <code><a href="com.html">com</a></code>(com.xyz): Center of Mass </li>
            
      <li>
        <code><a href="combine.sel.html">combine.sel</a></code>: Combine Atom Selections From PDB Structure </li>
            
      <li>
        <code><a href="dccm.enma.html">dccm.enma</a></code>: Cross-Correlation for Ensemble NMA (eNMA) </li>
            
      <li>
        <code><a href="dccm.mean.html">dccm.mean</a></code>:
</li>
            
      <li>
        <code><a href="dccm.nma.html">dccm.nma</a></code>: Dynamic Cross-Correlation from Normal Modes Analysis </li>
            
      <li>
        <code><a href="dccm.xyz.html">dccm.xyz</a></code>: DCCM: Dynamical Cross-Correlation Matrix </li>
            
      <li>
        <code><a href="deformation.nma.html">deformation.nma</a></code>: Deformation Analysis </li>
            
      <li>
        <code><a href="dssp.trj.html">dssp.trj</a></code>: Secondary Structure Analysis of Trajectories with DSSP  </li>
            
      <li>
        <code><a href="fluct.nma.html">fluct.nma</a></code>: NMA Fluctuations </li>
            
      <li>
        <code><a href="inner.prod.html">inner.prod</a></code>: Mass-weighted Inner Product </li>
            
      <li>
        <code><a href="is.pdb.html">is.pdb</a></code>:
</li>
            
      <li>
        <code><a href="is.select.html">is.select</a></code>:Is an Object of Class </li>
            
      <li>
        <code><a href="load.enmff.html">load.enmff</a></code>(ff.calpha, ff.calphax, ff.anm, ff.pfanm, ff.sdenm, ff.reach): ENM Force Field Loader </li>
            
      <li>
        <code><a href="mktrj.nma.html">mktrj.nma</a></code>: NMA Atomic Displacement Trajectory </li>
            
      <li>
        <code><a href="nma.html">nma</a></code>(build.hessian, print.nma): Normal Mode Analysis </li>
            
      <li>
        <code><a href="nma.pdbs.html">nma.pdbs</a></code>(print.enma): Ensemble Normal Mode Analysis </li>
            
      <li>
        <code><a href="normalize.vector.html">normalize.vector</a></code>: Mass-Weighted Normalized Vector </li>
            
      <li>
        <code><a href="pdb.annotate.html">pdb.annotate</a></code>: Get Customizable Annotations From PDB </li>
            
      <li>
        <code><a href="pdb2aln.html">pdb2aln</a></code>: Align a PDB structure to an existing alignment </li>
            
      <li>
        <code><a href="pdb2aln.ind.html">pdb2aln.ind</a></code>: Mapping between PDB atomic indices and alignment positions </li>
            
      <li>
        <code><a href="pdbfit.html">pdbfit</a></code>: PDB File Coordinate Superposition </li>
            
      <li>
        <code><a href="pdbs2pdb.html">pdbs2pdb</a></code>: PDBs to PDB Converter </li>
            
      <li>
        <code><a href="plot.enma.html">plot.enma</a></code>: Plot eNMA Results </li>
            
      <li>
        <code><a href="plot.nma.html">plot.nma</a></code>: Plot NMA Results </li>
            
      <li>
        <code><a href="plot.rmsip.html">plot.rmsip</a></code>: Plot RMSIP Results </li>
            
      <li>
        <code><a href="read.mol2.html">read.mol2</a></code>: Read MOL2 File </li>
            
      <li>
        <code><a href="sdENM.html">sdENM</a></code>: Index for the sdENM ff </li>
            
      <li>
        <code><a href="sse.bridges.html">sse.bridges</a></code>: SSE Backbone Hydrogen Bonding </li>
            
      <li>
        <code><a href="struct.aln.html">struct.aln</a></code>: Structure Alignment Of Two PDB Files </li>
            
      <li>
        <code><a href="view.dccm.html">view.dccm</a></code>: Visualization of Dynamic Cross-Correlation </li>
            
      <li>
        <code><a href="view.modes.html">view.modes</a></code>: Vector Field Visualization of Modes </li>
            
      <li>
        <code><a href="vmd.colors.html">vmd.colors</a></code>:
</li>
    
    </ul>
  </div>
  
  <div class="span3 offset1">
        
    <h2>Dependencies</h2>
    <ul>
      <li><strong>Depends</strong>: </li>
      <li><strong>Imports</strong>: </li>
      <li><strong>Suggests</strong>: ncdf, lattice, grid, bigmemory, multicore, XML, knitr</li>
      <li><strong>Extends</strong>: </li>
    </ul>
     
    
  </div>
</div>
      
      <footer>
      <p class='feedback'>What do you think of the documentation?  <a href='http://xinqyao.wufoo.com/forms/documentation-feedback-for-bio3d/def/field1='>Please let me know by filling out this short online survey</a>.</p>
<p class="pull-right"><a href="#">Back to top</a></p>
<p>Built by <a href="https://github.com/hadley/staticdocs">staticdocs</a>. Styled with <a href="http://twitter.github.com/bootstrap">bootstrap</a>.</p>
      </footer>
    </div>
  </body>
</html>
