<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>torsion.xyz. bio3d 2.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content=" Karim ElSawy ">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">

<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="http://thegrantlab.org/bio3d/index.html">bio3d 2.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Index</a></li>
          <li><a href="http://thegrantlab.org/bio3d/download/download.html">Download Bio3D</a></li>
          <li><a href="http://thegrantlab.org">The Grant Lab</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h1> Calculate Torsion/Dihedral Angles </h1>

<div class="row">
  <div class="span8">
    <h2>Usage</h2>
    <pre>torsion.xyz(xyz, atm.inc = 4)</pre>
    
    <h2>Arguments</h2>
    <dl>
      <dt>xyz</dt>
      <dd> a numeric vector of Cartisean coordinates. </dd>
      <dt>atm.inc</dt>
      <dd> a numeric value indicating the number of atoms to
    increment by between successive torsion evaluations (see below). </dd>
    </dl>
    
    <div class="Description">
      <h2>Description</h2>
      
      <p>Defined from the Cartesian coordinates of four successive atoms
  (A-B-C-D) the torsion or dihedral angle is calculated about an axis
  defined by the middle pair of atoms (B-C).</p>
  
    </div>

    <div class="Details">
      <h2>Details</h2>
      
      <p>The conformation of a polypeptide or nucleotide chain can be usefully
  described in terms of angles of internal rotation around its
  constituent bonds.</p>
  
      <p>If a system of four atoms A-B-C-D is projected onto a plane normal to
  bond B-C, the angle between the projection of A-B and the projection
  of C-D is described as the torsion angle of A and D about bond B-C.</p>
  
      <p>By convention angles are measured in the range -180 to +180, rather
  than from 0 to 360, with positive values defined to be in the
  clockwise direction.</p>
  
      <p>With <code>atm.inc=1</code>, torsion angles are calculated for each set of
  four successive atoms contained in <code>xyz</code> (i.e. moving along one
  atom, or three elements of <code>xyz</code>, between sucessive
  evaluations).  With <code>atm.inc=4</code>, torsion angles are calculated
  for each set of four successive non-overlapping atoms contained in
  <code>xyz</code> (i.e. moving along four atoms, or twelve elements of
  <code>xyz</code>, between sucessive evaluations).</p>
  
      <p></p>
  
    </div>

    <div class="Value">
      <h2>Value</h2>
      
      <p>A numeric vector of torsion angles.</p>
  
    </div>

    <div class="References">
      <h2>References</h2>
      
      <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
  
    </div>

    <div class="Note">
      <h2>Note</h2>
      
      <p>Contributions from Barry Grant.</p>
  
    </div>
    
    <h2 id="examples">Examples</h2>
    <pre class="examples"><div class='input'>## Calculate torsions for cis & trans conformers
xyz <- rbind(c(0,-0.5,0,1,0,0,1,1,0,0,1.5,0),
             c(0,-0.5,0,1,0,0,1,1,0,2,1.5,0)-3)
cis.tor   <- torsion.xyz( xyz[1,] )
trans.tor <- torsion.xyz( xyz[2,] )
apply(xyz, 1, torsion.xyz)
</div>
<div class='output'>[1]   0 180
</div>
<div class='input'>

plot(range(xyz), range(xyz), xlab="", ylab="", typ="n", axes=FALSE)
</div>
<p><img src='torsion.xyz-4.png' alt='' width='400' height='400' /></p>
<div class='input'>  apply(xyz, 1, function(x){
    lines(matrix(x, ncol=3, byrow=TRUE), lwd=4)
    points(matrix(x, ncol=3, byrow=TRUE), cex=2.5,
           bg="white", col="black", pch=21) } )
</div>
<p><img src='torsion.xyz-6.png' alt='' width='400' height='400' /></p>
<div class='output'>NULL
</div>
<div class='input'>  
  text( t(apply(xyz, 1, function(x){
    apply(matrix(x, ncol=3, byrow=TRUE)[c(2,3),], 2, mean) })),
       labels=c(0,180), adj=-0.5, col="red")
</div>
<p><img src='torsion.xyz-9.png' alt='' width='400' height='400' /></p>
<div class='input'>

##-- PDB torsion analysis
pdb <- read.pdb("1bg2")
</div>
<div class='output'>  Note: Accessing online PDB file
  HEADER    MOTOR PROTEIN                           04-JUN-98   1BG2               
</div>
<div class='input'>tor <- torsion.pdb(pdb)
## basic Ramachandran plot
plot(tor$phi, tor$psi)
</div>
<p><img src='torsion.xyz-13.png' alt='' width='400' height='400' /></p>
<div class='input'>

## torsion analysis of a single coordinate vector 
inds <- atom.select(pdb,"calpha")
</div>
<div class='output'>
Build selection from input string


 Using selection 'string' keyword shortcut: calpha = //////CA/ 

      segid  chain  resno  resid  eleno  elety
Stest ""     ""     ""     ""     ""     "CA" 
Natom "2527" "2527" "2527" "2527" "2527" "323"
 *  Selected a total of: 323 intersecting atoms  *
</div>
<div class='input'>tor.ca <- torsion.xyz(pdb$xyz[inds$xyz], atm.inc=3)

##-- Compare two PDBs to highlight interesting residues
aln <- read.fasta(system.file("examples/kif1a.fa",package="bio3d"))
m <- read.fasta.pdb(aln)
</div>
<div class='output'>pdb/seq: 1   name: http://www.rcsb.org/pdb/files/1bg2.pdb 
pdb/seq: 2   name: http://www.rcsb.org/pdb/files/1i6i.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 3   name: http://www.rcsb.org/pdb/files/1i5s.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 4   name: http://www.rcsb.org/pdb/files/2ncd.pdb 
</div>
<div class='input'>a <- torsion.xyz(m$xyz[1,],1)
b <- torsion.xyz(m$xyz[2,],1)
## Note the periodicity of torsion angles
d <- wrap.tor(a-b)
plot(m$resno[1,],d, typ="h")
</div>
<p><img src='torsion.xyz-19.png' alt='' width='400' height='400' /></p>
<div class='input'></div></pre>
  </div>
  <div class="span4">
    <!-- <ul>
      <li>torsion.xyz</li>
    </ul>
    <ul>
      <li> utilities </li>
    </ul> -->
      
    <h2>See also</h2>
     <code><a href='torsion.pdb.html'>torsion.pdb</a></code>, <code><a href='pca.tor.html'>pca.tor</a></code>,
  <code><a href='wrap.tor.html'>wrap.tor</a></code>, <code><a href='read.pdb.html'>read.pdb</a></code>, <code><a href='read.dcd.html'>read.dcd</a></code>.

    
    <h2>Author</h2>
     Karim ElSawy 
    
  </div>
</div>
      
      <footer>
      <p class='feedback'>What do you think of the documentation?  <a href='http://xinqyao.wufoo.com/forms/documentation-feedback-for-bio3d/def/field1=torsion.xyz'>Please let me know by filling out this short online survey</a>.</p>
<p class="pull-right"><a href="#">Back to top</a></p>
<p>Built by <a href="https://github.com/hadley/staticdocs">staticdocs</a>. Styled with <a href="http://twitter.github.com/bootstrap">bootstrap</a>.</p>
      </footer>
    </div>
  </body>
</html>
